jax/ad_util.py

```diff
@@ -43,13 +44,19 @@ def add_impl(xs, ys):
 def add_abstract(xs, ys):
   return lattice_join(xs, ys)
 
+def float0s_like(a):
+  return core.Float0Array(a.shape)
+
 jaxval_zeros_likers: Dict[type, Array] = {}
+jaxval_zeros_likers[dtypes.dtype(core.Float0Array)] = float0s_like
```

Comment:
I think `jaxval_zeros_likers` should map from a Python type (i.e. class) to the zeros-like function, rather than from dtype to the zeros like function. For example, we expect `numpy.ndarray` to be a key in this table but not all the numpy dtypes. That is, I think we want something like

```suggestion
jaxval_zeros_likers[core.Float0Array] = float0s_like
```